src/reanimated2/animation/repeat.ts

```diff
@@ -100,7 +108,14 @@ export const withRepeat = function <T extends AnimationObject>(
         now: Timestamp,
         previousAnimation: Animation<any> | null
       ): void {
-        animation.startValue = value;
+        // Detect re-render
+        const hasBeenInterrupted =
+          value !== initialAnimationBounds.start &&
+          value !== initialAnimationBounds.end;
```

Comment:
What about animations with easing which can go through end point while still in progress?

Reply:
It is close to impossible to force a re-render in the exact moment that an animation with easing goes through `toValue` point, but I am currently investigating whether we need comparing to `.end` whatsoever.